lib/reek/smells/attribute.rb

```diff
@@ -54,17 +48,35 @@ def examine_context(ctx)
       private
 
       def attributes_in(module_ctx)
+        result = Set.new
         module_ctx.local_nodes(:send) do |call_node|
-          if visibility_modifier?(call_node)
-            track_visibility(call_node)
-          elsif ATTR_DEFN_METHODS.include?(call_node.method_name)
-            call_node.arg_names.each do |arg|
-              @visiblity_tracker[arg] = @visiblity_mode
-              @result << [arg, call_node.line]
-            end
-          end
+          result += track_attributes(call_node)
         end
-        @result.select { |args| recorded_public_methods.include?(args[0]) }
+        result.select { |args| recorded_public_methods.include?(args[0]) }
+      end
+
+      def track_attributes(call_node)
+        if attribute_writer? call_node
+          return track_arguments call_node.args, call_node.line
+        end
+        track_visibility call_node if visibility_modifier? call_node
```

Comment:
Fantastic readability. If this was only everywhere like that in our codebase..;)